app/controllers/pages_controller.php

```diff
@@ -133,6 +133,12 @@ private function _homepageForMembers() {
         $latestMessages = $this->Wall->getLastMessages(5);
 
         $this->set('latestMessages', $latestMessages);
+
+        if (isset($this->Auth->user('settings')['hide_random_sentence'])) {
+            $this->set('hideRandomSentence', $this->Auth->user('settings')['hide_random_sentence']);
+        } else {
+            $this->set('hideRandomSentence', false);
+        }
```

Comment:
You can use `CurrentUser::getSetting('hide_random_sentence')` instead of `$this->Auth->user('settings')['hide_random_sentence'])`.

You can simplify a bit the code:
```
$hideRandomSentence = CurrentUser::getSetting('hide_random_sentence');
$this->set('hideRandomSentence', $hideRandomSentence);
```